src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -106,12 +98,20 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.marshal(toSave, fileWriter);
 
         } catch (IOException ioe) {
-            throw new StorageOperationException("Error writing to file: " + path);
+            checkIfFileIsReadOnly();
```

Comment:
No need to check for the writability explicitly. Exception already gives you that information. 
All you need to do is to handle it!